docs/20-Sortable-admin-list.md

```diff
@@ -0,0 +1,272 @@
+# Admin sortable list
+
+To sort an admin list by drag&drop
+
+![sortable-admin-list.png](img/sortable-admin-list.png)
+
+## How it works
+
+Specific templates are used by your admin to add a class to HTML table to init javascript drag&drop:
+- sonata/src/Ressources/views/CRUD/list_outer_rows_sortable_list.html.twig
+- sonata/src/Ressources/views/CRUD/list_sortable.html.twig
+
+When you drop the row of your list, an ajax call send all positions of your items list to a new route of your admin to save all positions
+
+## Installation
+
+You must install some javascript libraries to your project :
+ - "@shopify/draggable": "^1.0.0-beta.4"
+ - "superagent": "3.8.2"
+ 
+You must include css sonata/src/Ressources/public/css/sortableAdminList.css to your project
+ 
+You must call the sonata/src/Ressources/public/js/backoffice.js to init drag&drop if a sortable list is display
+
+You must declare a new service
+```
+<?xml version="1.0" encoding="UTF-8" ?>
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+
+        <!--SortAction-->
+        <service id="admin.sort_action" class="SortAction\SortAction">
+            <argument type="service" id="doctrine" />
+        </service>
+
+    </services>
+</container>
+```
+
+## How to implement drag&drop to your admin
+
+To use drag&drop to your entity admin, you must update your code like below
+
+### Position field
+
+You must add a field position to your entities
+(you can set an other name for this field if you want)
+
+```
+<?xml version="1.0" encoding="utf-8"?>
+<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping"
+                 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                 xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping
+                 http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
+
+    <entity name="CanalPlus\Bundle\AppBundle\Entity\YourEntity" table="your_entity_table">
+        <id name="id" type="integer" column="id">
+            <generator strategy="AUTO"/>
+        </id>
+
+        <field name="position" type="integer" column="position" nullable="true"/>
+        ...
+
+    </entity>
+</doctrine-mapping>
+```
+
+
+```
+class YourEntity
+{
+    ...
+
+    /**
+     * @var int
+     */
+    protected $position;
+
+    ...
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setPosition($position)
+    {
+        $this->position = $position;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPosition()
+    {
+        return $this->position;
+    }
+}
+```
+
+### Admin 
+
+You must add some config to your admin object :
```

Comment:
Could we provide a base admin class with those features enabled by default?

Reply:
Or a trait maybe?

Reply:
there are some methods like `configureListFields`, `configureRoutes` or `prePersist` that can be specific so they cannot be declared by a trait

Reply:
In this case, maybe a ``SortableAdmin`` abstract class extending the base ``Admin`` class?